### PR TITLE
Lower complexity of Form:isValid()

### PR DESCRIPTION
**Pull request by krzysiekpiasecki**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Replace #18391
- use Yoda condition,
- merge into 2.7 branch

Why?
- lower LOC
- less return expressions
- lower complexity
- more readable comparison

### CODE REVIEW

File: src/Symfony/Component/Form/Form.php [hunk 1]
```diff
@@ -766,11 +766,7 @@ public function isValid()
             return true;
         }
 
-        if (count($this->getErrors(true)) > 0) {
-            return false;
-        }
-
-        return true;
+        return 0 === count($this->getErrors(true));
```
Comment (stloyd) [inline src/Symfony/Component/Form/Form.php]: Why not just:
```php
return empty($this->getErrors(true));

Comment (fabpot) [inline src/Symfony/Component/Form/Form.php]: because being explicit is always better. We try to avoid using `empty` when possible.

Comment (HeahDude) [inline src/Symfony/Component/Form/Form.php]: @stloyd doesn't symfony standards prefer strict comparisons when possible ?

**Comment by HeahDude** [comment]
:+1:

File: src/Symfony/Component/Form/Form.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Form.php]: Also because getErrors returns a countable object, not an array.

**Comment by javiereguiluz** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @krzysiekpiasecki.

**Pull request closed without merging**